### DISCUSSION TITLE
doing 3D tracking based on an A-grid model output?

### DISCUSSION THREAD

**Discussion by keli559** [Parcels-code/Parcels#1244] [Q&A] [answered]
Learning how to do 3D tracking. However, the 3D tracking is based on a C-grid model output
Are there ways to do 3D tracking based on an A-grid model output? (meaning u v w are found at the same point)
thanks

**Comment by erikvansebille** [comment, marked as answer]
You can simply use the AdvectionRK4_3D Kernel. The only requirement is that your fieldset has a ‘W’ field, the grid type does not matter

**Comment by JamiePringle** [comment]
I know I am late in answering this -- but a word of caution. If you are using A grid data obtained by averaging a C-grid (e.g. the default Mercator Ocean output) you will find that the the A grid currents will run particles through land boundaries and the bottom. The land boundaries are only enforced on the C grid, not the averaged A grid. It is important to use the same grid as the model boundaries are implemented on. Copernicus will release C grid data if you ask them.  Jamie

**Reply by erikvansebille** [reply]
Yes, very good point, Jamie! This issue with ill boundary conditions in interpolated fields is also what the [Delandmeter and Van Sebille paper](https://www.geosci-model-dev.net/12/3571/2019/gmd-12-3571-2019.html) is (partially) about. 

And note that the uninterpolated data for Copernicus is also available via the portal of Mercator Ocean. E.g. https://www.mercator-ocean.eu/en/solutions-expertise/accessing-digital-data/ and https://www.mercator-ocean.fr/wp-content/uploads/products_pdf/MOI_GLO12.pdf

**Reply by flopesdematos** [reply]
In the context of 3D particle tracking using model outputs based on A-grid data obtained by averaging a C-grid, such as those available from Copernicus Marine Services, is there an easy-to-implement coding solution to prevent particles from behaving as "superparticles" capable of moving across the seafloor?

**Reply by JamiePringle** [reply]
not really. The problem with the A grid data is it will have velocity through boundaries. You can replace the particles back into the ocean by "magic" -- i.e. just teleport them back into the grid. I have not done this, but my graduate student has for particles with vertical behaviors (e.g. W=0), in which case similar problems can arise.  

Having said that, I strongly suggest using C grid data for C grid models. It will save you much pain in the future, and the new Mercator data downloading setup is much, much faster than the old downloading system. 

Jamie
